server/neptune/workflows/activities/terraform/root.go

```diff
@@ -20,6 +20,7 @@ type Root struct {
 	Trigger      Trigger
 	Rerun        bool
 	TrackedFiles []string
+	WorkflowMode WorkflowMode
```

Comment:
Few questions/thoughts:
1. WorkflowMode seems overloaded because we're also using it to identify default vs. platform mode,  I think i missed this in a past PR but I see there is PR and deploy, Just calling it out, but not sure what a better name would be.
2. Why is this hanging off the root vs. just part of the request overall? Doesn't seem like root specific metadata really. To me it's more like a parameter telling the workflow which steps to run through so it feels like it should be on the request, similar to deployment id.

Reply:
@nishkrishnan 
1. We call it WorkflowMode in gateway, I figured the concept of default vs. platform mode as we migrate onto Temporal and it would make better sense when in the TemporalWorker, we view these roots as PR vs Deploy modes. I could also change it to PR vs. Platform if that sounds more clear, but I was trying to just tie the name to what parent workflow spins up the Terraform workflow. What do you think?

2. Currently, gateway does hang its version of WorkflowMode off of the root, so I was following that behavior for simplicity (when implementing the signal on gateway's side I was just going to convert it to the new workflow mode version while passing it through). You are right that it doesn't have to be root specific metadata in the long term. I can move it to the workflow request.

Reply:
one extra thing to note is that we would also have to pass the workflow mode info in through state receiver/notifiers so that our CheckRunNotifier knows how to build the UI to include a row for the policy check step instead of the apply step.